docs/DeveloperGuide.adoc

```diff
@@ -279,104 +298,446 @@ Refer to the guide <<DevOps#, here>>.
-.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1. User requests to view list of transactions for a specific month.
+2. THRIFT displays transactions of that month.
++
+_Use case ends_
 
-_{More to be added}_
+[discrete]
+=== Use case: Tag a transaction
 
-[appendix]
-== Glossary
+*MSS*
+= *Extensions*
 
-[[mainstream-os]] Mainstream OS::
-Windows, Linux, Unix, OS-X
+1. User requests to tag a specific transaction.
+2. THRIFT processes the request and adds the tag to the specific transaction.
++
+_Use case ends_
+
+*Extensions*
+
+[none]
+* 1a. The command specified by the user is not valid.
++
+[none]
+** 1a1. THRIFT displays an invalid command error to user.
++
+_Use case ends_
+
+
+[discrete]
+=== Use case: Untag a transaction
 
-[[private-contact-detail]] Private contact detail::
-A contact detail that is not meant to be shared with others
+*MSS*
+
+1. User requests to untag a specific transaction.
+2. THRIFT processes the request and removes the tag from the specific transaction.
++
+_Use case ends_
+
+*Extensions*
+
+[none]
+* 1a. The parameters specified by the user are not valid.
++
+[none]
+** 1a1. THRIFT displays an invalid parameter error to the user and the transaction is not updated.
++
+_Use case ends_
+
+[discrete]
+=== Use case: Convert a transaction
+
+*MSS*
+
+1. User requests to convert the currency of a specific transaction to a new currency.
+2. THRIFT processes the request and updates specific transaction with the new value in the new currency.
++
+_Use case ends_
+
+*Extensions*
+
+[none]
+* 1a. The parameters specified by the user are not valid.
++
+[none]
+** 1a1. THRIFT displays an invalid parameter error to the user and the transaction is not updated.
++
+_Use case ends_
+
+[discrete]
+=== Use case: Convert a value
+
+*MSS*
+
+1. User requests to convert a value in SGD to a specific currency.
+2. THRIFT processes the request and displays the new value in the new currency.
++
+_Use case ends_
+
+*Extensions*
+
+[none]
+* 1a. The parameters specified by the user are not valid.
++
+[none]
+** 1a1. THRIFT displays an invalid parameter error to the user.
++
+_Use case ends_
+
+[appendix]
+== Non Functional Requirements
+
+. THRIFT should be able to run on any <<mainstream-os, mainstream OS>> as long as it has `Java 11` installed.
+. THRIFT should be able to hold up to 250 transactions without a noticeable drop in performance for typical usage.
+. A user with above average typing speed for <<regular-english-text, regular English text>> (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+. THRIFT should be able to run with or without internet connection.
+. THRIFT should work for a single user only.
+. THRIFT should not require user to install.
```

Comment:
It is possible to clarify this as 'THRIFT is portable and does not require an installer.'